Mathlib/Analysis/InnerProductSpace/JointEigenspace.lean

```diff
@@ -28,32 +31,48 @@ and a proof obligation that the basis vectors are eigenvectors.
 
 ## Tags
 
-self-adjoint operator, simultaneous eigenspaces, joint eigenspaces
+symmetric operator, simultaneous eigenspaces, joint eigenspaces
 
 -/
 
-variable {𝕜 E : Type*} [RCLike 𝕜]
-variable [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]
-
 open Module.End
 
 namespace LinearMap
 
 namespace IsSymmetric
 
-section Pair
+section CommRing
 
+variable {𝕜 E n m: Type*} [CommRing 𝕜] [AddCommGroup E] [Module 𝕜 E]
 variable {α : 𝕜} {A B : E →ₗ[𝕜] E}
 
-/--If a pair of operators commute, then the eigenspaces of one are invariant under the other.-/
+/-- If a pair of operators commute, then the eigenspaces of one are invariant under the other. -/
 theorem eigenspace_invariant_of_commute
     (hAB : A ∘ₗ B = B ∘ₗ A) (α : 𝕜) : ∀ v ∈ (eigenspace A α), (B v ∈ eigenspace A α) := by
   intro v hv
   rw [eigenspace, mem_ker, sub_apply, Module.algebraMap_end_apply, ← comp_apply A B v, hAB,
     comp_apply B A v, ← map_smul, ← map_sub, hv, map_zero] at *
 
-/--The simultaneous eigenspaces of a pair of commuting symmetric operators form an
-`OrthogonalFamily`.-/
+/-- The indexed infimum of eigenspaces of a commuting family of linear operators is
+invariant under each operator. -/
+theorem iInf_eigenspace_invariant_of_commute {T : n → E →ₗ[𝕜] E}
+    (hC : ∀ i j, T i ∘ₗ T j = T j ∘ₗ T i) (i : n) (γ : {x // x ≠ i} → 𝕜) ⦃v : E⦄
+    (hv : v ∈ ⨅ j, eigenspace (Subtype.restrict (· ≠ i) T j) (γ j)) :
+    T i v ∈ ⨅ j, eigenspace (Subtype.restrict (· ≠ i) T j) (γ j) := by
+  simp only [Submodule.mem_iInf] at hv ⊢
+  exact fun j ↦ eigenspace_invariant_of_commute (hC j i) (γ j) v (hv j)
+
+end CommRing
+
+open Submodule
+
+section RCLike
+
+variable {𝕜 E n m: Type*} [RCLike 𝕜] [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]
+variable {α : 𝕜} {A B : E →ₗ[𝕜] E}
```

Comment:
any reason not to do this and remove the corresponding arguments from the theorems below?
```suggestion
variable {α : 𝕜} {A B : E →ₗ[𝕜] E} {T : n → E →ₗ[𝕜] E}
```

Reply:
I think this is totally reasonable.

Reply:
I will try this and then, I guess, if Lean complains below because it can't see T, I can make it explicit in each statement.

Reply:
Yes, it weirdly breaks some things below (the first result requiring T doesn't need T fed in explicitly, but in the very last theorem, when we CALL this first result, if T isn't explicit in the first result, the theorem breaks). I'm puzzled by this because I would have expected Lean to be able to recover `T` from `hT` whenever.